### PR TITLE
Sema: Make "@retroactive attribute does not apply" a warning

### PR DESCRIPTION
**Pull request by tshortli**
Follow up to https://github.com/apple/swift/pull/71302.

Resolves rdar://122590301

### CODE REVIEW

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Review by harlanhaskins — APPROVED** [review]

File: include/swift/AST/DiagnosticsSema.def [hunk 1]
```diff
@@ -2544,9 +2544,9 @@ ERROR(retroactive_not_in_extension_inheritance_clause,none,
       "'retroactive' attribute only applies in inheritance clauses in "
       "extensions", ())
 
-ERROR(retroactive_attr_does_not_apply,none,
-      "'retroactive' attribute does not apply; %0 is declared in this module",
-      (Identifier))
+WARNING(retroactive_attr_does_not_apply,none,
```
Comment (DougGregor) [inline include/swift/AST/DiagnosticsSema.def:2547]: This is specified in the proposal to be an error. How about we leave it as an error and use `warnUntilSwiftVersion(6)` to stage it in?

**Comment by harlanhaskins** [comment]
Disabling the auto-merge per Doug’s comment — I’m okay with that but I also understand the urgency with which we need to downgrade this warning

**Comment by tshortli** [comment]
@swift-ci please smoke test

File: include/swift/AST/DiagnosticsSema.def (hunk 1 above)
Comment (tshortli) [reply include/swift/AST/DiagnosticsSema.def]: Done

**Pull request merged**